### PR TITLE
Data Preprocessing

### PR DESCRIPTION
**Pull request by dixslyf**
Tasks:
- [x] Split the dataset into test and train sets
- [x] Feature scaling
- [x] Encode nominal and ordinal features
- [ ] ~(Maybe) Dimensionality reduction~

### CODE REVIEW

File: data_training.ipynb [hunk 1]
```diff
@@ -0,0 +1,305 @@
...
+    "    \"family_history_with_overweight\",\n",
+    "    \"FAVC\",\n",
+    "    \"CAEC\",\n",
+    "    \"SMOKE\",\n",
+    "    \"SCC\",\n",
+    "    \"CALC\",\n",
+    "    \"MTRANS\"\n",
+    "]\n",
+    "\n",
+    "ohe = OneHotEncoder(sparse_output=False)\n",
```
Comment (dixslyf) [inline data_training.ipynb:247]: May want to specify `drop="first"`. I'll need to read more about it.

Comment (dixslyf) [reply data_training.ipynb]: It's not so straightforward. It depends on the model we're going to use and whether regularization is applied. If regularization is used, then we shouldn't drop the first column. For now, I'm going to drop the first column — we can revisit this later.

File: data_training.ipynb [hunk 2]
```diff
@@ -0,0 +1,305 @@
...
+   "source": [
+    "import os\n",
+    "\n",
+    "import pandas as pd\n",
+    "from sklearn.preprocessing import StandardScaler, LabelEncoder, OneHotEncoder\n",
+    "from sklearn.compose import make_column_transformer\n",
+    "from sklearn.model_selection import train_test_split\n",
+    "import numpy as np"
+   ]
+  },
```
Comment (Cyrof) [inline data_training.ipynb:34]: - The imports are clear and well-organised.
- `make_column_transformer` from `sklearn.compose` is imported but not used. Consider removing it if it's not needed.

File: data_training.ipynb [hunk 3]
```diff
@@ -0,0 +1,305 @@
...
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "le = LabelEncoder()\n",
+    "le.fit(ob_df[\"NObeyesdad\"])\n",
+    "\n",
+    "y_train[:] = le.transform(y_train)\n",
+    "y_test[:] = le.transform(y_test)"
+   ]
+  },
```
Comment (Cyrof) [inline data_training.ipynb:200]: - Label encoding is appropriately used for the target variable. 
- Ensure the `y_train[:]` and `y_test[:]` correctly reference and update the series; alternatively, consider using `.loc` for clarity.

File: data_training.ipynb
Comment (Cyrof) [inline data_training.ipynb:1]: ## Final Suggestions
- Consider adding comments or markdown cells explaining the rationale behind each step for better readability.

**Review by Cyrof — CHANGES_REQUESTED** [review]

File: data_training.ipynb (hunk 2 above)
Comment (dixslyf) [reply data_training.ipynb:34]: Good catch.

**Review by Cyrof — APPROVED** [review]
